packages/clay-shared/src/Portal.tsx

```diff
@@ -4,18 +4,40 @@
  * SPDX-License-Identifier: BSD-3-Clause
  */
 
-import React, {useEffect, useRef} from 'react';
+import React, {createContext, useContext, useEffect, useRef} from 'react';
 import {createPortal} from 'react-dom';
 
+const ClayPortalContext = createContext<React.RefObject<Element> | null>(null);
+
+ClayPortalContext.displayName = 'ClayPortalContext';
+
 export const ClayPortal: React.FunctionComponent<
-	React.HTMLAttributes<HTMLDivElement>
-> = ({children}) => {
-	const portalRef = useRef(
-		typeof document !== 'undefined' && document.createElement('div')
-	);
-	const elToMountTo = typeof document !== 'undefined' && document.body;
+	React.HTMLAttributes<HTMLDivElement> & {
+		/**
+		 * Ref of element to render portal into.
+		 */
+		containerRef?: React.RefObject<Element>;
+
+		/**
+		 * Ref of element to render nested portals into.
+		 */
+		subPortalRef?: React.RefObject<Element>;
+	}
+> = ({children, containerRef, subPortalRef}) => {
+	const parentPortalRef = useContext(ClayPortalContext);
+	const portalRef = useRef(document.createElement('div'));
```

Comment:
You can check if there is `document` here, I was having problems with it because it was being rendered on `next.clayui.com` and was breaking the site build (log https://app.netlify.com/sites/next-clayui/deploys/5d5595f8912d82502b05f511), because at build time there is no `document`.

```javascript
typeof document !== 'undefined' && document.createElement('div')
```